### DISCUSSION TITLE
switching FMS version

### DISCUSSION THREAD

**Discussion by jiandewang** [ufs-community/ufs-weather-model#2310] [Q&A]
I am testing one of GFDL's updated MOM6 code but it failed on FMS related issue. Currenly we are using FMS 2023.04 in UWM. I need to test UWM with FMS 2024.01. Can someone explain to me on how to replace FMS version here ? Thanks

**Comment by junwang-noaa** [comment]
You have to install and update the module file 
https://github.com/ufs-community/ufs-weather-model/blob/develop/modulefiles/ufs_common.lua#L14

@hang-lei-noaa is testing fms/2024.01 on wcoss2.

**Comment by jiandewang** [comment]
@junwang-noaa thanks for the information. Let me reach out to Hang

**Comment by jiandewang** [comment]
I changed /lfs/h2/emc/couple/noscrub/Jiande.Wang/MOM6-update/20240531/ufs-weather-model/modulefiles/ufs_common.lua

 {["fms"]             = "2024.01"},

but in my compiling log on wcoss2
/lfs/h2/emc/ptmp/jiande.wang/FV3_RT/rt_119486/compile_s2s_intel/err, 
I still see it is using 2023.04
5) udunits/2.2.28           22) fms/2023.04

is there anything I missed here ?

**Comment by jkbk2004** [comment]
@jiandewang Do you want to test FMS 2023.04.01 on RDHPCS side: hercules?

**Comment by jiandewang** [comment]
@jkbk2004 "FMS 2023.04.01" ? is there typo in your response ? what I need is 2024.01 or later

**Comment by Hang-Lei-NOAA** [comment]
@Jiande Wang - NOAA Affiliate ***@***.***> I can help you with
testing on wcoss2 and acorn

**Comment by Hang-Lei-NOAA** [comment]
This is the installation on acorn:

Please test the library at
***@***.***:~/noscrub/works/nco_wcoss2> module use
/lfs/h1/emc/nceplibs/noscrub/hpc-stack/libs/hpc-stack/modulefiles/mpi/intel/19.1.3.304/cray-mpich/8.1.9
***@***.***:~/noscrub/works/nco_wcoss2> module load fms/2024.01
***@***.***:~/noscrub/works/nco_wcoss2> module show fms/2024.01
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

 /lfs/h1/emc/nceplibs/noscrub/hpc-stack/libs/hpc-stack/modulefiles/mpi/intel/19.1.3.304/cray-mpich/8.1.9/fms/2024.01.lua:
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
help([[]])
conflict("fms")
setenv("FMS_ROOT","/lfs/h1/emc/nceplibs/noscrub/hpc-stack/libs/hpc-stack/intel-19.1.3.304/cray-mpich-8.1.9/fms/2024.01")
setenv("FMS_VERSION","2024.01")
whatis("Name: fms")
whatis("Version: 2024.01")
whatis("Category: library")
whatis("Description: GFDL FMS library")

Let me know if you have any problem in your model, which may also need an
old env.
Hang

**Comment by jiandewang** [comment]
@Hang-Lei-NOAA thanks for the detail information. I will reach out to you if I encounter difficult
